internal/command/plan_test.go

```diff
@@ -15,7 +15,79 @@ import (
 	"github.com/google/go-cmp/cmp"
 )
 
-func TestPlan(t *testing.T) {
+func TestPlanJSON(t *testing.T) {
+	svr := getHttptestServer()
+	defer svr.Close()
+
+	cfg := getConfig()
+	cfg.ServerBaseUrl = svr.URL
+
+	if err := cfg.ValidateForPlan(); err != nil {
+		t.Errorf("Invalid config: %v", err)
+	}
+
+	ctx := context.Background()
+
+	// By default command.Run writes to os.Stdout.
+	// Replace with a string buffer here so we can test the command output.
+	var buf bytes.Buffer
+	command.SetPlanWriter(&buf)
+
+	// This is the method under test
+	err := command.Plan(ctx, cfg, "", command.PlanOutputJSON, "")
+
+	if err != nil {
+		t.Errorf("command.Plan(...) error = %v", err)
+	}
+
+	want := `{"BUILDKITE_TEST_ENGINE_PLAN_IDENTIFIER":"facecafe","BUILDKITE_TEST_ENGINE_PARALLELISM":"42"}
+`
+	got := buf.String()
+
+	if diff := cmp.Diff(want, got); diff != "" {
+		t.Errorf("command.Plan(...) diff = %s", diff)
+	}
+}
+
+func TestPlanPipelineUpload(t *testing.T) {
+	svr := getHttptestServer()
+	defer svr.Close()
+
+	cfg := getConfig()
+	cfg.ServerBaseUrl = svr.URL
+
+	if err := cfg.ValidateForPlan(); err != nil {
+		t.Errorf("Invalid config: %v", err)
+	}
+
+	ctx := context.Background()
+
+	// By default command.Run writes to os.Stdout.
+	// Replace with a string buffer here so we can test the command output.
+	var buf bytes.Buffer
+	command.SetPlanWriter(&buf)
```

Comment:
I was writing some tests for fallback mode and noticed that this syntax only looks at planWriter output and not all of stdout. I wonder if there is a more robust approach?